linkschkr.go

```diff
@@ -0,0 +1,127 @@
+package linkschkr
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"golang.org/x/net/html"
+)
+
+type Work struct {
+	site   string
+	result chan *Result
+}
+
+type Result struct {
+	site      string
+	up        bool
+	extraURIs []string
+	workerID  int
+}
+
+func Crawler(ID int, c <-chan *Work) {
+	for {
+		work := <-c
```

Comment:
Maybe this should be a `select` statement? That way, we could take a context, and also listen to its `Done` channel so we can be cancelled.